pkg/provider/provider.go

```diff
@@ -376,16 +385,33 @@ func (k *kubeProvider) Configure(_ context.Context, req *pulumirpc.ConfigureRequ
 func (k *kubeProvider) Invoke(ctx context.Context,
 	req *pulumirpc.InvokeRequest) (*pulumirpc.InvokeResponse, error) {
 
+	// Important: Some invoke logic is intended to run during preview, and the Kubernetes provider
+	// inputs may not have resolved yet. Rather than returning an error here, any invoke logic must
+	// not assume that a cluster is accessible.
+	if k.clusterUnreachable {
+		glog.V(3).Infof(
+			"configured Kubernetes cluster is unreachable. Invoke call logic may operate in a degraded state.")
+	}
```

Comment:
Do we benefit from having this at this level - or should it just live closer to the point where we actually take a dependency on the cluster - where we can report more explicitly what that implies.

Reply:
Yeah, I think that makes more sense. I'll leave a note here about the implications, but defer warnings to the point in the logic where it matters.